packages/flutter/lib/src/material/navigation_rail.dart

```diff
@@ -749,6 +768,10 @@ class _RailDestination extends StatelessWidget {
           Semantics(
             label: indexLabel,
           ),
+          if (verticalSpacing != 0 || itemSpacing != 0)
+          SizedBox(
+            height: itemSpacing != 0 ? itemSpacing : verticalSpacing,
+          ),
```

Comment:
I don't think this has the right effect. For example, in the example app in the api docs (https://master-api.flutter.dev/flutter/material/NavigationRail-class.html), you will not see any difference until you have a itemSpacing or verticalSpacing value greater than ~70-80 because this just overlays the SizedBox on top of each item/icon in the navigation rail.

Reply:
Yes that's true. I guess this issue would no longer exist after making the suggested implementation of wrapping items in Padding widget.